### DISCUSSION TITLE
editing vtk, finding the E-field

### DISCUSSION THREAD

**Discussion by new99** [thliebig/openEMS-Project#44] [Q&A Usage]
I ran simulations and got the E-field by saving it in a vtk file, let's call it E0. Then I simulated with the addition of an object, let's call it E. I did not change the grid size, number of iterations, or iteration step. **How do I get |E/E0|?** 
I tried Paraview, maybe I didn't look hard enough.

`from paraview.simple import *`
`reader = OpenDataFile("tmp_mod/Et_0000001990.vtr")`
`print(reader.PointData.keys())`
`E0 = reader.PointData.GetArray('E-Field')`
`print(E0)`
`print(E0.GetRange())`

I get the result:

['E-Field']
Array: E-Field
(-0.02497160993516445, 0.01990814507007599)

Expected a three-dimensional array.

**Comment by thliebig** [comment]
I have not used paraview in python. So far I only used the GUI. Try to open both sets of files and in theory there should also be a math node to do the E/E0...
But first I would look at the data itself

**Reply by new99** [reply]
simulated two cases:
1) without an object (E0)
![E0_full](https://user-images.githubusercontent.com/16485403/224390478-66834ba9-a048-4663-83fe-fba8b0c34bd8.png)

2) with an object (E)
![E_full](https://user-images.githubusercontent.com/16485403/224390546-47b4a943-51b5-41e8-aa5c-159c4ad7a2e1.png)

**Reply by new99** [reply]
Is that what you meant?

**Reply by new99** [reply]
I don't understand what buttons to press in Paraview to get |E/E0|.

**Reply by thliebig** [reply]
Well there is a "Filters" Menu with e.g. the "Calculator" filter. I think this one can do this. But I haven't used it in many years...

**Reply by new99** [reply]
I renamed E-field to E0-filed
![image](https://user-images.githubusercontent.com/16485403/224417039-947da409-e286-49c0-83fc-38bd8ba08855.png)

grouped using Group Datasets
![image](https://user-images.githubusercontent.com/16485403/224417401-8e288dfd-d5cb-4c21-9619-d470e8242aad.png)

Calculator works only with E-field or E0-field.
![image](https://user-images.githubusercontent.com/16485403/224417696-1f7754a6-0fce-413e-b47f-571ef72e0b6d.png)

dividing the E-field by the E0-field does not work
![image](https://user-images.githubusercontent.com/16485403/224417989-59c557b8-f56c-4932-8ed9-f4f2126d9ad5.png)

**Reply by new99** [reply]
with "E-Field_X" "E-Field_Y" "E-Field_Z" also tried

**Reply by thliebig** [reply]
Well I guess then you have to store/dump the field into the hdf5 file format. Read these files with python or Octave, do whatever array op you desire and either plot directly in python/Octave or write an vtk file to review in paraview

**Reply by new99** [reply]
So you can't do that with Paraview?

**Reply by thliebig** [reply]
I think it should be possible... but depends if you can figure it out and what is maybe easier at the end...
